docs: added connector documentation (#2786)
FuelLabs/fuels-ts: apps/docs/src/guide/wallets/connectors.md

Author: Please could I get some opinions on how to separate this? Also where it could lie in the nav? :)

It requires two arguments:

- `contractId` (`string`)
- `abi` ([`FuelABI`](../../api/Account/index.md#fuelabi)).

It will return a promise that will resolve to `true` if the ABI is successfully added; otherwise `false`.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-addABI{ts:line-numbers}

#### `getABI`

The `getABI` method is used to get the ABI information that is sorted about a contract.

It requires a single argument:

- `contractId` (`string`)

Returns a promise that resolves to the ABI information (as a [`FuelABI`](../../api/Account/index.md#fuelabi)) or `null` if the data is unavailable.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-getABI{ts:line-numbers}

#### `hasABI`

The `hasABI` method checks if the ABI information is available for a contract.

It requires a single argument:

- `contractId` (`string`)

Returns a promise that resolves to `true` if the ABI information is available; otherwise `false`.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-hasABI{ts:line-numbers}

> ## Fuel SDK

Reviewer: ```suggestion
## Connectors Manager
```

Author: 45c4f54ee6055e5a1aae0d537f0fc69636bf3419

> The `Fuel` manager provides the interface for interaction with the TS SDK and the broader Fuel ecosystem.

Reviewer: ```suggestion
The TS SDK exports the `Fuel` class, which serves as the connectors manager. This class provides the interface for interacting with the TS SDK and the broader Fuel ecosystem.
```

Author: 45c4f54ee6055e5a1aae0d537f0fc69636bf3419

- `contractId` (`string`)

Returns a promise that resolves to the ABI information (as a [`FuelABI`](../../api/Account/index.md#fuelabi)) or `null` if the data is unavailable.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-getABI{ts:line-numbers}

#### `hasABI`

The `hasABI` method checks if the ABI information is available for a contract.

It requires a single argument:

- `contractId` (`string`)

Returns a promise that resolves to `true` if the ABI information is available; otherwise `false`.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-hasABI{ts:line-numbers}

## Connectors Manager

The TS SDK exports the `Fuel` class, which serves as the connectors manager. This class provides the interface for interacting with the TS SDK and the broader Fuel ecosystem.

It can be instantiated as follows:

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-instantiation-1{ts:line-numbers}

### Options

Several options can be passed to the `Fuel` connector manager:

- [`connectors`](#connectors)
- [`storage`](#storage)
- [`targetObject`](#targetobject)

#### `connectors`

> The `connectors` option provides a list of connectors with which the `Fuel` connector manager can interact. The manager interacts with the connectors, which in turn handle communication with the respective wallet. You can find a list of all the connectors in our [`FuelLabs/fuel-connectors`](https://github.com/FuelLabs/fuel-connectors).

Author: @petertonysmith94 create an issue:
- requesting for a list of all the currently supported connectors and their related wallet install links.
- update this line + link to reference that page.

> You can find a list of all the connectors in our [`FuelLabs/fuel-connectors`](https://github.com/FuelLabs/fuel-connectors).

Reviewer 2: Is the intention here to create an issue and handle this on a separate PR? I'm not seeing the issue

Author: - https://github.com/FuelLabs/fuels-ts/issues/2834

> By using these connectors, developers can simplify wallet integration and enable smooth user interactions within their applications.

Reviewer: ```md
<!-- before -->
Fuel Wallet Connectors provide a standardized interface to integrate multiple wallets with your DApps.

By using these connectors, developers can simplify wallet integration and enable smooth user interactions within their applications.

<!-- after -->
Fuel Wallet Connectors offer a standardized interface to integrate multiple wallets with your DApps, simplifying wallet integration and ensuring smooth user interactions.
```

Author: [2b37b6f](https://github.com/FuelLabs/fuels-ts/pull/2786/commits/2b37b6f2493b84416158c50c599b8c438968535d)

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-addAsset{ts:line-numbers}

#### `addAssets`

The `addAssets` method adds multiple asset metadata to the connector.

It requires a single argument:

- `assets` (an Array of [`Asset`](../../api/Account/index.md#asset)).

Returns a promise that resolves to `true` if the assets are successfully added; otherwise, resolves to `false`.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-addAssets{ts:line-numbers}

#### `addNetwork`

The `addNetwork` method starts the add network flow for the current connection.

It requires a single argument:

- `networkUrl` (`string`)

Returns a promise that resolves to `true` if the network is successfully added; otherwise, `false`.

It should throw an error if the network is not available or the network already exists.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-addNetwork{ts:line-numbers}

#### `networks`

The `networks` method returns a list of all the networks available for the current connection.

> Returns a promise that resolves to an array of networks (see [`Network`](../../api/Account/index.md#network)) that are available on the network.

Reviewer 2: ```suggestion
Returns a promise that resolves to an array of available networks (see [`Network`](../../api/Account/index.md#network)).
```

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-networks{ts:line-numbers}

#### `currentNetwork`

The `currentNetwork` method will return the current network that is connected.

It will return a promise that will resolve to the current network (see [`Network`](../../api/Account/index.md#network)).

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-currentNetwork{ts:line-numbers}

#### `selectNetwork`

> The `selectNetwork` method is used to request the user to select a network for the current connection.

Reviewer 3: ```suggestion
The `selectNetwork` method requests the user to select a network for the current connection.
```

It requires a single argument:

- `network` ([`Network`](../../api/Account/index.md#network))

> It will return a promise that will resolve to `true` if the network is successfully selected, otherwise `false`.

Reviewer 3: ```suggestion
It will return a promise that resolves to `true` if the network is successfully selected; otherwise, it will return `false`.
```

It should throw an error if the network is not available or the network does _not_ exist.

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-selectNetwork{ts:line-numbers}

#### `addABI`

> The `addABI` method is used to add ABI information about a contract to the connector. This operation does no require an authorized connection.

Reviewer 3: ```suggestion
The `addABI` method adds ABI information about a contract to the connector. This operation does not require an authorized connection.
```

It requires two arguments:

- `contractId` (`string`)
- `abi` ([`FuelABI`](../../api/Account/index.md#fuelabi)).

> It will return a promise that will resolve to `true` if the ABI is successfully added, otherwise `false`.

Reviewer 3: ```suggestion
It will return a promise that will resolve to `true` if the ABI is successfully added; otherwise `false`.
```

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-addABI{ts:line-numbers}

#### `getABI`

The `getABI` method is used to get the ABI information that is sorted about a contract.

It requires a single argument:

- `contractId` (`string`)

> It will return a promise that will resolve to the ABI information (as a [`FuelABI`](../../api/Account/index.md#fuelabi)) or `null` if the data is unavailable.

Reviewer 3: ```suggestion
Returns a promise that resolves to the ABI information (as a [`FuelABI`](../../api/Account/index.md#fuelabi)) or `null` if the data is unavailable.
```

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-getABI{ts:line-numbers}

#### `hasABI`

> The `hasABI` method is used to check if the ABI information is available for a contract.

Reviewer 3: ```suggestion
The `hasABI` method checks if the ABI information is available for a contract.
```

It requires a single argument:

- `contractId` (`string`)

> It will return a promise that will resolve to `true` if the ABI information is available, otherwise `false`.

Reviewer 3: ```suggestion
Returns a promise that resolves to `true` if the ABI information is available; otherwise `false`.
```

<<< @/../../../packages/account/src/connectors/fuel-connector.ts#fuel-connector-method-hasABI{ts:line-numbers}

## Fuel SDK

> The `Fuel` manager provides the interface for interaction with the TS SDK and the wider Fuel ecosystem.

Reviewer 3: ```suggestion
The `Fuel` manager provides the interface for interaction with the TS SDK and the broader Fuel ecosystem.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

It can be instantiated as follows:

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-instantiation-1{ts:line-numbers}

### Options

> There are a number of options that can be passed to the `Fuel` manager:

Reviewer 3: ```suggestion
There are several options that can be passed to the `Fuel` manager:
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

- [`connectors`](#connectors)
- [`storage`](#storage)

> - [`targetObject`](#targetObject)

Reviewer 3: ```
Anchor #targetObject does not exist in connectors.md
```

Author: 07f133b61698161d2a4cedd7fd1255ec9836b142

#### `connectors`

> The `connectors` option is used to provide a list of connectors that the `Fuel` manager can interact with. The manager will interact with the connectors that will in turn handle the communication with the respective wallet. You can find a list of all the connectors in our [`FuelLabs/fuel-connectors`](https://github.com/FuelLabs/fuel-connectors). Below you will find a snippet of using the current default connectors that are provided by the `fuel-connectors` package.

Reviewer 3: ```suggestion
The `connectors` option provides a list of connectors with which the `Fuel` manager can interact. The manager interacts with the connectors, which in turn handle communication with the respective wallet. You can find a list of all the connectors in our [`FuelLabs/fuel-connectors`](https://github.com/FuelLabs/fuel-connectors). Below, you will find a snippet of using the current default connectors provided by the `fuel-connectors` package.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-options-connectors{ts:line-numbers}

#### `storage`

> The `storage` is used internally to store the current connector state. This can be overriden as follows, by passing an instance that extends the `StorageAbstract` class.

Reviewer 3: ```suggestion
The `storage` is used internally to store the current connector state. It can be overridden by passing an instance that extends the `StorageAbstract` class.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-options-storage-memory{ts:line-numbers}

The default behavior will use `LocalStorage` if the `window` is available:

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-options-storage-local{ts:line-numbers}

#### `targetObject`

> The `targetObject` is used to provide a target for the `Fuel` manager to interact with. Used for registering events and can be overridden as follows:

Reviewer 3: ```suggestion
The `targetObject` provides a target with which the `Fuel` manager can interact. Used for registering events and can be overridden as follows:
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../docs-snippets/src/guide/wallets/connectors.test.ts#fuel-options-target-object{ts:line-numbers}

### Methods

> The `Fuel` manager provides a number of methods to interact with the SDK:

Reviewer 3: ```suggestion
The `Fuel` manager provides several methods to interact with the SDK:
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

#### All methods from connectors

> The `Fuel` manager provides all the [methods](#methods) that are available from the connectors that are currently connected. This means that you can interact with the current connector as if you were interacting with the `Fuel` manager directly.

Reviewer 3: ```suggestion
The `Fuel` manager provides all the [methods](#methods) available from the connected connectors. Thus, you can interact with the current connector as if you were interacting with the `Fuel` manager directly.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

> If there is no current connector available or connected, it will throw an error.

Reviewer 3: ```suggestion
If no current connector is available or connected, it will throw an error.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

#### `connectors`

> The `connectors` method is used to get the current list of _installed_ and _connected_ connectors that are currently available.

Reviewer 3: ```suggestion
The `connectors` method gets the current list of _installed_ and _connected_ connectors.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-connectors{ts:line-numbers}

#### `getConnector`

> The `getConnector` method is used to resolve a connector by its name. This is useful for getting a specific connector that you want to interact with. If the connector is not found, it will return `null`.

Reviewer 3: ```suggestion
The `getConnector` method resolves a connector by its name. This is useful for finding a specific connector with which to interact. If the connector is not found, it will return `null`.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-getConnector{ts:line-numbers}

#### `hasConnector`

The `hasConnector` method will return `true` under the following conditions:

- There is a current connector that is connected.
- A connector is connected within two seconds of calling the method.

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-hasConnector{ts:line-numbers}

#### `selectConnector`

> The `selectConnector` method accepts a connector name and will return `true` when it is _available_ and _connected_. Otherwise, if not found or is unavailable, it will return `false`.

Reviewer 3: ```suggestion
The `selectConnector` method accepts a connector name and will return `true` when it is _available_ and _connected_. Otherwise, if not found or unavailable, it will return `false`.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-selectConnector{ts:line-numbers}

#### `currentConnector`

The `currentConnector` method will return the current connector that is connected or if one is available and connected, otherwise it'll return `null` or `undefined`.

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-currentConnector{ts:line-numbers}

#### `getWallet`

The `getWallet` method accepts an address (string or instance) as the first parameter and a provider or network as the second parameter. It will return an `Account` instance for the given address (providing it is valid).

> The provider or network will default to the current network if not provided. When a provider cannot be resolved, it will throw an error of code [`INVALID_PROVIDER`](../errors/index.md).

Reviewer 2: ```suggestion
The provider or network will default to the current network if not provided. When a provider cannot be resolved, it will throw an [`INVALID_PROVIDER`](../errors/index.md) error.

> The `getWallet` method accepts an address (string or instance) as the first parameter and a provider or network as the second parameter. It will return an `Account` instance for the given address (providing the address is valid).

Reviewer 3: ```suggestion
The `getWallet` method accepts an address (string or instance) as the first parameter and a provider or network as the second parameter. It will return an `Account` instance for the given address (providing it is valid).
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

> If the provider or network is not provided, it will default to the current network. When a provider is unable to be resolved, it will throw an error of code `INVALID_PROVIDER`.

Reviewer 3: ```suggestion
The provider or network will default to the current network if not provided. When a provider cannot be resolved, it will throw an error of code `INVALID_PROVIDER`.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-getWallet{ts:line-numbers}

#### `clean`

The `clean` method removes all the data currently stored in the [`storage`](#storage) instance.

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-clean{ts:line-numbers}

#### `unsubscribe`

The `unsubscribe` method removes all currently registered event listeners.

<<< @/../../../packages/account/src/connectors/fuel.ts#fuel-sdk-method-unsubscribe{ts:line-numbers}

#### `destroy`

> The `destroy` method removes all the event listeners and cleans the storage (a combination of `clean` and `unsubscribe`).

Reviewer 2: ```suggestion
The `destroy` method unsubscribes from all the event listeners and clears the storage.
```

> The `clean` method removes all the data that is currently stored in the [`storage`](#storage) instance.

Reviewer 3: ```suggestion
The `clean` method removes all the data currently stored in the [`storage`](#storage) instance.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

> The `unsubscribe` method is used to remove all the event listeners that are currently registered.

Reviewer 3: ```suggestion
The `unsubscribe` method removes all currently registered event listeners.
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90

> The `destroy` method is used to remove all the event listeners and clean the storage (a combination of `clean` and `unsubscribe`).

Reviewer 3: ```suggestion
The `destroy` method removes all the event listeners and cleans the storage (a combination of `clean` and `unsubscribe`).
```

Author: 09e0871ef593c3cdbf7aa799b437996ef2232e90
